microbit-v2/Cargo.toml

```diff
@@ -34,3 +34,6 @@ path = "src/lib.rs"
 path = "../microbit-common"
 features = ["v2"]
 version = "=0.13.0"
+
+[features]
+embedded-hal-02 = ["microbit-common/embedded-hal-02"]
```

Comment:
I would recommend not making embedded-hal 0.2 an optional feature (yet).
There's still a lot of the driver ecosystem that has not updated to 1.0, and if your target audience is newbies you want to give them the least amount of things they must change.

Reply:
I'd like to let people avoid both the extra dependency which may be unnecessary, and the confusing error messages (e.g. in hints to add missing use statements for traits) that come from depending on two different versions of the same library.

Reply:
Fair enough. I hadn't seen those particular errors first-hand in a while, so I did up a quick test and I agree - those messages are going to be tricky for newbies to parse and understand.
Optional 0.2 hal does look like a better option than the alternative.